gw_eccentricity/eccDefinition.py

```diff
@@ -251,24 +297,178 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
         self.t_for_omega22_average = None
         self.omega22_average = None
 
-        if "hlm_zeroecc" in self.dataDict:
-            self.compute_res_amp_and_omega22()
+        # compute residual data
+        if "amplm_zeroecc" in self.dataDict and "omegalm_zeroecc" in self.dataDict:
+            self.compute_res_amp22_and_res_omega22()
 
     def get_recognized_dataDict_keys(self):
         """Get the list of recognized keys in dataDict."""
         list_of_keys = [
             "t",                # time array of waveform modes
             "hlm",              # Dict of eccentric waveform modes
+            "amplm",            # Dict of amplitude of eccentric waveform modes
+            "phaselm",          # Dict of phase of eccentric waveform modes
+            "omegalm",          # Dict of omega of eccentric waveform modes
             "t_zeroecc",        # time array of quasicircular waveform
             "hlm_zeroecc",      # Dict of quasicircular waveform modes
+            "amplm_zeroecc",    # Dict of amplitude of quasicircular waveform modes
+            "phaselm_zeroecc",  # Dict of phase of quasicircular waveform modes
+            "omegalm_zeroecc",  # Dict of omega of quasicircular waveform modes
         ]
         return list_of_keys
 
-    def truncate_dataDict_if_necessary(self,
-                                       dataDict,
-                                       num_orbits_to_exclude_before_merger,
-                                       extra_kwargs):
-        """Truncate dataDict if "num_orbits_to_exclude_before_merger" is not None.
+    def get_amp_phase_omega_data(self, dataDict):
+        """Get the dictionary of amplitude, omega and phase from user dataDict.
+
+        The `dataDict` provided by the user can contain any of the data
+        corresponding to the keys in `get_recognized_dataDict_keys`. To compute
+        eccentricity and mean anomaly, we need amplitude, phase and omega. In
+        this function, we get these data from `dataDict` and create a new
+        dictionary that contains only the amplitude, phase and omega of the
+        available modes.
+
+        For example, if `dataDict` has only the complex hlm modes, this
+        functions uses the hlm modes to create a dictionary containing the
+        amplitude, phase and omega by decomposing the hlm modes. Afterwards,
+        only these waveform data in the new dataDict is used for all the
+        further computations.
```

Comment:
"all the further computations" --> "all further computations"
Good to run a grammar check.

Reply:
done